src/Symfony/Component/Validator/Constraints/DefaultPasswordStrengthEstimator.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+final class DefaultPasswordStrengthEstimator implements PasswordStrengthEstimatorInterface
+{
+    /**
+     * @var non-empty-array<array{score: int, minEntropy: int}>
+     */
+    private const CRITERIA = [
+        ['score' => 4, 'minEntropy' => 120],
+        ['score' => 3, 'minEntropy' => 100],
+        ['score' => 2, 'minEntropy' => 80],
+        ['score' => 1, 'minEntropy' => 60],
+        ['score' => 0, 'minEntropy' => 0],
+    ];
+
+    /**
+     * @param string[] $restrictedData
```

Comment:
what is restrictedData about? deniedPasswords?

Reply:
In the meantime I split the contraint in 2. I created a `BannedWords` constraint that have all the logic.
Could be used in other contexts.